cmake/ActsCompilerOptions.cmake

```diff
@@ -4,12 +4,16 @@ if (NOT CMAKE_BUILD_TYPE)
   message(STATUS "Setting default build type: ${CMAKE_BUILD_TYPE}")
 endif() 
 
-set(ACTS_CXX_FLAGS "-Wall -Wextra -Wpedantic -Wshadow -Wno-unused-local-typedefs")
+set(ACTS_CXX_FLAGS "-Wall -Wextra -Wpedantic -Wshadow -Wfloat-conversion -Wno-unused-local-typedefs")
 set(ACTS_CXX_FLAGS_DEBUG "--coverage")
 set(ACTS_CXX_FLAGS_MINSIZEREL "")
 set(ACTS_CXX_FLAGS_RELEASE "")
 set(ACTS_CXX_FLAGS_RELWITHDEBINFO "")
 
+#if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang") 
+#  set(ACTS_CXX_FLAGS "${ACTS_CXX_FLAGS} -Wfloat-conversion")
+#endif()
+
```

Comment:
What was this about?

Reply:
Hmm I think clang and gcc behave differently on this (see also the GCC builds failing)
I think we wanted to test that back then...

Reply:
I think gcc does not allow an that fine-grained warning configuration, so from my point of view it would be an option to just enable the warnings on clang builds